### PR TITLE
Add req argument to custom success/error

### PR DESCRIPTION
**Pull request by trevorr**
Fixes https://github.com/pinojs/pino-http/issues/208

### CODE REVIEW

**Review by mcollina — CHANGES_REQUESTED** [review]
I think we should keep the order `req, res` even if it will require a semver-major bump.

**Comment by trevorr** [comment]
No problem. Should `err` still be first in that case?

**Comment by mcollina** [comment]
If it's there, yes!

**Comment by trevorr** [comment]
I swapped the order where I added `req`. Should I also add to `customLogLevel`? Does it still make sense to have `err` in a different order between `customErrorMessage` and `customLogLevel`?

```ts
customLogLevel?: ((res: ServerResponse, error: Error) => pino.LevelWithSilent) | undefined;
customReceivedMessage?: ((req: IncomingMessage, res: ServerResponse) => string) | undefined;
customSuccessMessage?: ((req: IncomingMessage, res: ServerResponse) => string) | undefined;
customErrorMessage?: ((error: Error, req: IncomingMessage, res: ServerResponse) => string) | undefined;
```

**Comment by trevorr** [comment]
I also noticed that the docs have `customProps` while the Typescript types have `reqCustomProps` (and the code supports both). Perhaps if we're bumping the major version, this is a good time to rename one of them?

**Comment by mcollina** [comment]
Go ahead and fix customLogLevel and resolve the problem with the types! Thanks for the help!

**Comment by trevorr** [comment]
Ok, we now have this:

```ts
customLogLevel?: ((req: IncomingMessage, res: ServerResponse, error: Error) => pino.LevelWithSilent) | undefined;
customReceivedMessage?: ((req: IncomingMessage, res: ServerResponse) => string) | undefined;
customSuccessMessage?: ((req: IncomingMessage, res: ServerResponse) => string) | undefined;
customErrorMessage?: ((error: Error, req: IncomingMessage, res: ServerResponse) => string) | undefined;
customProps?: ((req: IncomingMessage, res: ServerResponse) => object) | undefined;
```

`customErrorMessage` still has `error` first, but I think it makes sense in that case.

**Review by mcollina — APPROVED** [review]
lgtm

File: README.md [hunk 1]
```diff
@@ -96,18 +96,18 @@ $ node example.js | pino-pretty
...
 * `customReceivedMessage`: set to a `function (req, res) => { /* returns message string */ }` This function will be invoked at each request received, setting "msg" property to returned string. If not set, nothing value will be used.
-* `customSuccessMessage`: set to a `function (res) => { /* returns message string */ }` This function will be invoked at each successful response, setting "msg" property to returned string. If not set, default value will be used.
-* `customErrorMessage`: set to a `function (err, res) => { /* returns message string */ }` This function will be invoked at each failed response, setting "msg" property to returned string. If not set, default value will be used.
+* `customSuccessMessage`: set to a `function (req, res) => { /* returns message string */ }` This function will be invoked at each successful response, setting "msg" property to returned string. If not set, default value will be used.
+* `customErrorMessage`: set to a `function (err, req, res) => { /* returns message string */ }` This function will be invoked at each failed response, setting "msg" property to returned string. If not set, default value will be used.
```
Comment (jsumners) [inline README.md:107]: This is inconsistent with the new signature for `customLogLevel`.

Given that this PR is semver-major, I would just make every signature `aFunc({ req, res, err })`.

Comment (trevorr) [reply README.md]: Ok, done.

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Comment by trevorr** [comment]
Do you know when you might merge and release this?

**Comment by mcollina** [comment]
Monday or Tuesday if @jsumners  doesn't beat me to it

**Comment by jsumners** [comment]
bobs-burgers-gene

**Pull request merged**